packages/edit-post/src/hooks/components/index.js

```diff
@@ -11,3 +18,78 @@ addFilter(
 	'core/edit-post/replace-media-upload',
 	replaceMediaUpload
 );
+
+function getMediaSourceUrlBySizeSlug( media, slug ) {
+	return (
+		media?.media_details?.sizes?.[ slug ]?.source_url || media?.source_url
+	);
+}
+
+const addFeaturedImageToolbarItem = createHigherOrderComponent(
+	( BlockEdit ) => {
+		return ( props ) => {
+			const {
+				attributesContext = [],
+				attributes,
+				setAttributes,
+				name: blockName,
+			} = props;
+
+			const { useFeaturedImage } = attributes;
+
+			const { sizeSlug } = attributes;
+			const [ featuredImage ] = useEntityProp(
+				'postType',
+				'post',
+				'featured_media'
+			);
+
+			const media = useSelect(
+				( select ) =>
+					featuredImage &&
+					select( coreStore ).getMedia( featuredImage, {
+						context: 'view',
+					} ),
+				[ featuredImage ]
+			);
+			const mediaUrl = getMediaSourceUrlBySizeSlug( media, sizeSlug );
+
+			if ( useFeaturedImage ) {
+				attributesContext.push( { url: mediaUrl } );
```

Comment:
This synchronizes data from post to the block. Awesome! My mental model for this feature is "link this cover block to the featured image" so I would also like any changes to the block image be reflected in the post featured image, kind of like two way data sync.

Reply:
You are right, so is my mental model. Yet, this is, in my opinion, a next step once we stabilize the API to bind blocks to contextual data, next step being changing contextual data from blocks.